packages/flutter_tools/lib/src/windows/windows_version_validator.dart

```diff
@@ -16,14 +19,36 @@ const List<String> kUnsupportedVersions = <String>[
 /// (ie. 10.5.4123)
 const String kWindowsOSVersionSemVerPattern = r'([0-9]+)\.([0-9]+)\.([0-9\.]+)';
 
+/// Regex pattern for identifying a running instance of the Topaz OFD process.
+/// This is a known process that interferes with the build toolchain.
+/// See https://github.com/flutter/flutter/issues/121366
+const String kCoreProcessPattern = r'Topaz OFD\\Warsaw\\core\.exe';
+
 /// Validator for supported Windows host machine operating system version.
 class WindowsVersionValidator extends DoctorValidator {
   const WindowsVersionValidator({
     required OperatingSystemUtils operatingSystemUtils,
+    required ProcessLister processLister,
   })  : _operatingSystemUtils = operatingSystemUtils,
+        _processLister = processLister,
         super('Windows Version');
 
   final OperatingSystemUtils _operatingSystemUtils;
+  final ProcessLister _processLister;
+
+  Future<ValidationResult> _topazScan() async {
+      final ProcessResult getProcessesResult = await _processLister.getProcessesWithPath();
+      if (getProcessesResult.exitCode != 0) {
+        return const ValidationResult(ValidationType.missing, <ValidationMessage>[ValidationMessage.hint('Get-Process failed to complete')]);
+      }
+      final RegExp topazRegex = RegExp(kCoreProcessPattern, caseSensitive: false,  multiLine: true);
+      final String processes = getProcessesResult.stdout as String;
+      final bool topazFound = topazRegex.hasMatch(processes);
```

Comment:
Would it make sense to do a string contains instead of a regex match?

Reply:
I suppose we could pass it a slightly more convoluted pattern in order to use that method. I think it is best that we make the test case-insensitive and perhaps should use a regex shorthand for the space character.

Reply:
Since this contains only ASCII characters, you can use lowered strings to do a case-insensitive string comparison. Matching only the ASCII space (`U+0020`) seems acceptable.

Feel free to upvote this Dart request for `String.equalsIgnoreCase`: https://github.com/dart-lang/sdk/issues/54418 😅

Reply:
Do you have any other concerns on this change?